test/metabase/prometheus_test.clj

```diff
@@ -0,0 +1,146 @@
+(ns metabase.prometheus-test
+  (:require [clj-http.client :as http]
+            [clojure.set :as set]
+            [clojure.string :as str]
+            [clojure.test :refer :all]
+            [iapetos.registry :as registry]
+            [metabase.prometheus :as prometheus]
+            [metabase.test.fixtures :as fixtures]
+            [metabase.troubleshooting :as troubleshooting]))
+
+;; ensure a handler to instrument with jetty_stats and a db so the c3p0 stats have at least one connection
+(use-fixtures :once (fixtures/initialize :db :web-server))
+
+(def ^:private common-metrics
+  "Common metric types from the registry"
+  #{"c3p0_max_pool_size"
+    "c3p0_min_pool_size"
+    "c3p0_num_busy_connections"
+    "c3p0_num_connections"
+    "c3p0_num_idle_connections"
+    "jetty_async_dispatches_total"
+    "jetty_async_requests_total"
+    "jetty_async_requests_waiting"
+    "jetty_async_requests_waiting_max"
+    "jetty_dispatched_active"
+    "jetty_dispatched_active_max"
+    "jetty_dispatched_time_max"
+    "jetty_dispatched_time_seconds_total"
+    "jetty_dispatched_total"
+    "jetty_expires_total"
+    "jetty_request_time_max_seconds"
+    "jetty_requests_active"
+    "jetty_requests_active_max"
+    "jetty_requests_total"
+    "jetty_responses_bytes_total"
+    "jetty_responses_total"
+    "jetty_stats_seconds"
+    "jvm_gc_collection_seconds_count"
+    "jvm_gc_collection_seconds_sum"
+    "jvm_memory_bytes_committed"
+    "jvm_memory_bytes_init"
+    "jvm_memory_bytes_max"
+    "jvm_memory_bytes_used"
+    "jvm_memory_objects_pending_finalization"
+    "jvm_memory_pool_bytes_committed"
+    "jvm_memory_pool_bytes_init"
+    "jvm_memory_pool_bytes_max"
+    "jvm_memory_pool_bytes_used"
+    "jvm_memory_pool_collection_committed_bytes"
+    "jvm_memory_pool_collection_init_bytes"
+    "jvm_memory_pool_collection_max_bytes"
+    "jvm_memory_pool_collection_used_bytes"
+    "jvm_threads_current"
+    "jvm_threads_daemon"
+    "jvm_threads_deadlocked"
+    "jvm_threads_deadlocked_monitor"
+    "jvm_threads_peak"
+    "jvm_threads_started_total"
+    "jvm_threads_state"
+    "process_cpu_seconds_total"
+    "process_max_fds"
+    "process_open_fds"
+    "process_start_time_seconds"
+    "jetty_request_time_seconds_total"})
+
+(defn metric-tags
```

Comment:
Should this and `metric-lines` be private?

Reply:
I've updated that. In general I don't care about test namespaces because there are no consumers that need to be directed to the api of a system or datastructure.